project/driver_controls/src/driver_devices.c

```diff
@@ -0,0 +1,23 @@
+#include <stddef.h>
+
+#include "driver_devices.h"
+#include "gpio_it.h"
+
+void driver_controls_init() {
+  gpio_init();
+  interrupt_init();
+  gpio_it_init();
+}
+
+void driver_controls_add_device(Device *device) {
```

Comment:
Why does `add_device` not seem to add a device? It looks more like an init function to me.